photon-core/src/main/java/org/photonvision/vision/objects/AppleObjectDetector.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) Photon Vision.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.photonvision.vision.objects;
+
+import com.photonvision.apple.SwiftArena;
+import java.lang.foreign.MemorySegment;
+import java.lang.ref.Cleaner;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.opencv.core.Mat;
+import org.opencv.core.Rect2d;
+import org.opencv.core.Scalar;
+import org.opencv.core.Size;
+import org.opencv.imgproc.Imgproc;
+import org.photonvision.common.logging.LogGroup;
+import org.photonvision.common.logging.Logger;
+import org.photonvision.vision.pipe.impl.NeuralNetworkPipeResult;
+
+/**
+ * Manages an object detector using Apple's CoreML and Vision framework backend.
+ *
+ * <p>This detector uses Apple's Vision framework which provides automatic image preprocessing
+ * (resizing, cropping) and built-in NMS (Non-Maximum Suppression).
+ */
+public class AppleObjectDetector implements ObjectDetector {
+    private static final Logger logger = new Logger(AppleObjectDetector.class, LogGroup.General);
+
+    /** Cleaner instance to release the detector when it goes out of scope */
+    private final Cleaner cleaner = Cleaner.create();
```

Comment:
Other classes use Releaseable because we don't have control over when this cleanup happens. Prefer to manually destroy objects rather than relying on GC to do this